### PR TITLE
improve tests of L1CrossDomainMessenger_Test

### PR DESCRIPTION
**Pull request by AmadiMichael**
improve tests of L1CrossDomainMessenger_Test

### CODE REVIEW

File: packages/contracts-bedrock/test/L1/L1CrossDomainMessenger.t.sol [hunk 1]
```diff
@@ -173,21 +174,75 @@ contract L1CrossDomainMessenger_Test is CommonTest {
         assertEq(l1CrossDomainMessenger.failedMessages(hash), false);
     }
 
+    /// @dev Tests that relayMessage reverts if caller is optimismPortal and the value sent does not match the amount
+    /// and if a failedMessage is attempted to be replayed via the optimismPortal
+    /// signified.
+    function test_relayMessage_callerIsOptimismPortalChecks_reverts() external {
```
Comment (smartcontracts) [inline packages/contracts-bedrock/test/L1/L1CrossDomainMessenger.t.sol]: Let's break this into three tests

Comment (AmadiMichael) [reply packages/contracts-bedrock/test/L1/L1CrossDomainMessenger.t.sol]: fixed

File: packages/contracts-bedrock/test/L1/L1CrossDomainMessenger.t.sol [hunk 2]
```diff
@@ -173,21 +174,86 @@ contract L1CrossDomainMessenger_Test is CommonTest {
         assertEq(l1CrossDomainMessenger.failedMessages(hash), false);
     }
 
+    /// @dev Tests that relayMessage reverts if caller is optimismPortal and the value sent does not match the amount
+    function test_relayMessage_callerIsOptimismPortalAndValueDoesNotMatchAmount_reverts() external {
+        // set the target to be the OptimismPortal
```
Comment (smartcontracts) [inline packages/contracts-bedrock/test/L1/L1CrossDomainMessenger.t.sol]: Comment is wrong here

File: packages/contracts-bedrock/test/L1/L1CrossDomainMessenger.t.sol [hunk 3]
```diff
@@ -173,21 +174,86 @@ contract L1CrossDomainMessenger_Test is CommonTest {
...
+        address target = alice;
+        address sender = Predeploys.L2_CROSS_DOMAIN_MESSENGER;
+        bytes memory message = hex"1111";
+
+        // set the value of op.l2Sender() to be the L2 Cross Domain Messenger.
+        vm.store(address(optimismPortal), bytes32(senderSlotIndex), bytes32(abi.encode(sender)));
+
+        // cannot send a message as optimism portal but amount does not match msg.value
```
Comment (smartcontracts) [inline packages/contracts-bedrock/test/L1/L1CrossDomainMessenger.t.sol]: Better comment is probably something like "Correctly sending as OptimismPortal but amount does not match msg.value"

File: packages/contracts-bedrock/test/L1/L1CrossDomainMessenger.t.sol (hunk 2 above)
Comment (AmadiMichael) [reply packages/contracts-bedrock/test/L1/L1CrossDomainMessenger.t.sol]: fixed

File: packages/contracts-bedrock/test/L1/L1CrossDomainMessenger.t.sol (hunk 3 above)
Comment (AmadiMichael) [reply packages/contracts-bedrock/test/L1/L1CrossDomainMessenger.t.sol]: fixed

File: packages/contracts-bedrock/test/L1/L1CrossDomainMessenger.t.sol [hunk 4]
```diff
@@ -173,21 +174,86 @@ contract L1CrossDomainMessenger_Test is CommonTest {
         assertEq(l1CrossDomainMessenger.failedMessages(hash), false);
     }
 
+    /// @dev Tests that relayMessage reverts if caller is optimismPortal and the value sent does not match the amount
+    function test_relayMessage_callerIsOptimismPortalAndValueDoesNotMatchAmount_reverts() external {
```
Comment (smartcontracts) [inline packages/contracts-bedrock/test/L1/L1CrossDomainMessenger.t.sol]:
```suggestion
    function test_relayMessage_fromOtherMessengerValueMismatch_reverts() external {
```

File: packages/contracts-bedrock/test/L1/L1CrossDomainMessenger.t.sol [hunk 5]
```diff
@@ -173,21 +174,86 @@ contract L1CrossDomainMessenger_Test is CommonTest {
...
+        // set the target to be the alice
```
Comment (smartcontracts) [inline packages/contracts-bedrock/test/L1/L1CrossDomainMessenger.t.sol]: Can remove this comment

File: packages/contracts-bedrock/test/L1/L1CrossDomainMessenger.t.sol [hunk 6]
```diff
@@ -173,21 +174,86 @@ contract L1CrossDomainMessenger_Test is CommonTest {
...
+        address target = alice;
+        address sender = Predeploys.L2_CROSS_DOMAIN_MESSENGER;
+        bytes memory message = hex"1111";
+
+        // set the value of op.l2Sender() to be the L2 Cross Domain Messenger.
```
Comment (smartcontracts) [inline packages/contracts-bedrock/test/L1/L1CrossDomainMessenger.t.sol]:
```suggestion
        // set the value of op.l2Sender() to be the L2CrossDomainMessenger.
```

File: packages/contracts-bedrock/test/L1/L1CrossDomainMessenger.t.sol [hunk 7]
```diff
@@ -173,21 +174,86 @@ contract L1CrossDomainMessenger_Test is CommonTest {
...
+        vm.deal(address(optimismPortal), 10 ether);
+        vm.prank(address(optimismPortal));
+        vm.expectRevert(stdError.assertionError);
+        l1CrossDomainMessenger.relayMessage{ value: 10 ether }(
+            Encoding.encodeVersionedNonce({ _nonce: 0, _version: 1 }), sender, target, 9 ether, 0, message
+        );
+    }
+
+    /// @dev Tests that relayMessage reverts if a failed message is attempted to be replayed via the optimismPortal
+    function test_relayMessage_callerIsOptimismPortalAndFailedMessageReplay_reverts() external {
```
Comment (smartcontracts) [inline packages/contracts-bedrock/test/L1/L1CrossDomainMessenger.t.sol]:
```suggestion
    function test_relayMessage_fromOtherMessengerFailedMessageReplay_reverts() external {
```

File: packages/contracts-bedrock/test/L1/L1CrossDomainMessenger.t.sol [hunk 8]
```diff
@@ -173,21 +174,86 @@ contract L1CrossDomainMessenger_Test is CommonTest {
...
+        // set the target to be the OptimismPortal
```
Comment (smartcontracts) [inline packages/contracts-bedrock/test/L1/L1CrossDomainMessenger.t.sol]: Can remove this comment

File: packages/contracts-bedrock/test/L1/L1CrossDomainMessenger.t.sol [hunk 9]
```diff
@@ -173,21 +174,86 @@ contract L1CrossDomainMessenger_Test is CommonTest {
...
+        vm.prank(address(optimismPortal));
+        vm.expectRevert(stdError.assertionError);
+        l1CrossDomainMessenger.relayMessage(
+            Encoding.encodeVersionedNonce({ _nonce: 0, _version: 1 }), sender, target, 0, 0, message
+        );
+    }
+
     /// @dev Tests that relayMessage reverts if attempting to relay a message
     ///      sent to an L1 system contract.
     function test_relayMessage_toSystemContract_reverts() external {
```
Comment (smartcontracts) [inline packages/contracts-bedrock/test/L1/L1CrossDomainMessenger.t.sol]: Same comment as other PR to break this into two tests

File: packages/contracts-bedrock/test/L1/L1CrossDomainMessenger.t.sol [hunk 10]
```diff
@@ -173,21 +174,86 @@ contract L1CrossDomainMessenger_Test is CommonTest {
...
+        vm.prank(address(optimismPortal));
+        vm.expectRevert("CrossDomainMessenger: cannot send message to blocked system address");
+        l1CrossDomainMessenger.relayMessage(
+            Encoding.encodeVersionedNonce({ _nonce: 0, _version: 1 }), sender, address(optimismPortal), 0, 0, message
         );
+    }
+
+    /// @dev Tests that the relayMessage function reverts if the message called by non-optimismPortal but not a failed
+    /// message
+    function test_relayMessage_nonFailedMessageCalledByNonOtherMessenger_reverts() external {
```
Comment (smartcontracts) [inline packages/contracts-bedrock/test/L1/L1CrossDomainMessenger.t.sol]:
```suggestion
    function test_relayMessage_relayingNewMessageByExternalUser_reverts() external {
```

**Review by smartcontracts — APPROVED** [review]

**Pull request merged**